src/View/ViewableData.php

```diff
@@ -237,13 +233,45 @@ public function setField($field, $value)
         // prior to PHP 8.2 support ViewableData::setField() simply used `$this->field = $value;`
         // so the following logic essentially mimics this behaviour, though without the use
         // of now deprecated dynamic properties
-        if (property_exists($this, $field)) {
+        if ($this->isAccessibleProperty($field)) {
             $this->$field = $value;
         }
         $this->data[$field] = $value;
         return $this;
     }
 
+    /**
+     * Returns true if a method exists for the current class which isn't private.
+     * Also returns true for private methods if $this is ViewableData (not a subclass)
+     */
+    private function isAccessibleMethod(string $method): bool
+    {
+        if (!method_exists($this, $method)) {
+            return false;
+        }
+        if (static::class === self::class) {
```

Comment:
I may (probably am) misunderstanding this, though won't `self::class` always be `ViewableData::class`?

Is this what we're trying to do here?
```
if (!$reflectionMethod->isPrivate()) {
  return true;
} else {
  // method is private
  if (<method is defined on the class that called isAccessibleMethod(), or a trait that's been applied to that class>) {
    return true;
  } else {
    return false;
  }
}
```

Reply:
> won't `self::class` always be `ViewableData::class`?

Yes. So we're checking if the class of the object is `ViewableData` (and not some subclass).

What we're doing is:

1. Returning `false` if the method just simply doesn't exist. It's not accessible if it doesn't exist.
2. Returning `true` if the method exists and the static class is `ViewableData` - it can always access its own methods.
3. Returning `false` if the method exists, but the static class is _not_ `ViewableData`, and the method is private. `ViewableData` can't access methods which are private and defined in subclasses.
4. Returning `true` if the method exists, the static class is _not_ `ViewableData`, but the method is _not_ private. `ViewableData` can access protected and public methods of its subclasses.

The unit tests cover all of these bases, I believe.

Reply:
Ah my bad, forgot that isAccessibleMethod() is a private method itself so doesn't need to work on subclasses